diff --git a/configure b/configure
index 281751a..3b1f649 100755
--- a/configure
+++ b/configure
@@ -42413,7 +42413,7 @@ char BZ2_bzlibVersion ();
 
 #endif
 int
-main ()
+main (void)
 {
 return BZ2_bzlibVersion ();
   ;
@@ -42474,9 +42474,11 @@ else
 /* end confdefs.h.  */
 
 #ifdef HAVE_BZLIB_H
+#include <stdlib.h> // for exit
+#include <string.h> // for strcmp
 #include <bzlib.h>
 #endif
-int main() {
+int main(void) {
     char *ver = BZ2_bzlibVersion();
     exit(strcmp(ver, "1.0.6") < 0);
 }
@@ -43184,6 +43186,7 @@ else
   cat confdefs.h - <<_ACEOF >conftest.$ac_ext
 /* end confdefs.h.  */
 
+#include <stdlib.h>
 #include <string.h>
 #include <curl/curl.h>
 int main()

diff --git a/tools/config.guess b/tools/config.guess
index abcdef0..1234567 100755
--- a/tools/config.guess
+++ b/tools/config.guess
@@ -150,6 +150,13 @@ LIBC=gnu
 	#endif
 	EOF
 	eval `$CC_FOR_BUILD -E $dummy.c 2>/dev/null | grep '^LIBC' | sed 's, ,,g'`
+
+	# If ldd exists, use it to detect musl libc.
+	if command -v ldd >/dev/null && \
+		ldd --version 2>&1 | grep -q ^musl
+	then
+	    LIBC=musl
+	fi
 	;;
 esac
 
